### PR TITLE
Add tracking to header title, links and breadcrumb

### PR DESCRIPTION
**Pull request by danmitchell-**
### Context
As part of the GaaP programme we want to be able to track more events 

### Changes proposed in this pull request
Add the generic script from https://github.com/alphagov/gaap-analytics

### Guidance to review
Check data attributes match ORJ

### CODE REVIEW

File: app/views/layouts/application.html.haml [hunk 1]
```diff
@@ -30,9 +30,9 @@
             = link_to "Documentation", "https://registers-docs.cloudapps.digital/", target: :blank, rel: "noopener noreferrer"
 
 - content_for :head do
-  = stylesheet_link_tag 'application'
+  = stylesheet_link_tag 'application' if Rails.env.production?
```
Comment (karlbaker02) [inline app/views/layouts/application.html.haml]: Should this be on the `layouts/analytics` partial still?

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -13,3 +13,5 @@
 //= require jquery
 //= require jquery_ujs
 //= require_tree .
+
+window.GAAP.analytics.init();
```
Comment (karlbaker02) [inline app/assets/javascripts/application.js]: Do we need to require the `gaap-analytics.js` file in here too?

Comment (danmitchell-) [reply app/assets/javascripts/application.js]: `//= require_tree` brings in everything from the application directory

Comment (danmitchell-) [reply app/assets/javascripts/application.js]: You can remove `//= require_tree .` and specify the files individually when you want to control the load order.

**Pull request merged**